website/source/docs/agent/configuration/index.html.md

```diff
@@ -0,0 +1,254 @@
+---
+layout: "docs"
+page_title: "Agent Configuration"
+sidebar_current: "docs-agent-configuration"
+description: |-
+  Learn about the configuration options available for the Nomad agent.
+---
+
+# Agent Configuration
+
+Nomad agents have a variety of parameters that can be specified via
+configuration files or command-line flags. Configuration files are written in
+[HCL][hcl]. Nomad can read and combine parameters from multiple configuration
+files or directories to configure the Nomad agent.
+
+## Load Order and Merging
+
+The Nomad agent supports multiple configuration files, which can be provided
+using the `-config` CLI flag. The flag can accept either a file or folder:
+
+```shell
+$ nomad agent -config=server.conf -config=/etc/nomad -config=extra.json
+```
+
+This will load configuration from `server.conf`, from `.hcl` and `.json` files
+under `/etc/nomad`, and finally from `extra.json`. Files are loaded and merged
+in lexicographical order. Directories are not loaded recursively.
+
+As each file is processed, its contents are merged into the existing
+configuration. When merging, any non-empty values from the latest config file
+will append or replace parameters in the current configuration. An empty value
+means `""` for strings, `0` for integer or float values, and `false` for
+booleans. Since empty values are ignored you cannot disable an parameter like
+`server` mode once you've enabled it.
+
+Here is an example Nomad agent configuration that runs in both client and server
+mode.
+
+```hcl
+bind_addr = "0.0.0.0"
+data_dir  = "/var/lib/nomad"
+
+advertise {
+  # We need to specify our host's IP because we can't
+  # advertise 0.0.0.0 to other nodes in our cluster.
+  rpc = "1.2.3.4:4647"
+}
+
+server {
+  enabled          = true
+  bootstrap_expect = 3
+}
+
+client {
+  enabled       = true
+  network_speed = 10
+  options {
+    "driver.raw_exec.enable" = "1"
+  }
+}
+
+consul {
+  address = "1.2.3.4:8500"
+}
+
+atlas {
+  infrastructure = "hashicorp/mars"
+  token          = "atlas.v1.AFE84330943"
+}
+```
+
+~> Note that it is strongly recommended **not** to operate a node as both
+`client` and `server`, although this is supported to simplify development and
+testing.
+
+## General Parameters
+
+- `addresses` `(Addresses: see below)` - Specifies the bind address for
+  individual network services. Any values configured in this stanza take
+  precedence over the default [bind_addr](#bind_addr). If provided, **all
```

Comment:
That's my experience. If you specify this:

```
addresses {
  http = "1.2.3.4"
}
```

it sets rcp and serf addresses to `""`